core/ipc/sink/kafka/client/src/main/java/org/opennms/core/ipc/sink/kafka/client/KafkaRemoteMessageDispatcherFactory.java

```diff
@@ -89,12 +96,27 @@
                     Thread.currentThread().interrupt();
                     break;
                 } catch (ExecutionException e) {
-                    LOG.warn("Timeout occured while sending message to topic {}, it will be attempted again.", topic);
+                    // Timeout typically happens when Kafka is Offline or it didn't initialize yet.
+                    // For this case keep sending the message until it delivers, will cause sink messages to buffer.
+                    if (e.getCause() != null && e.getCause() instanceof TimeoutException) {
+                        LOG.warn("Timeout occured while sending message to topic {}, it will be attempted again.", topic, e);
```

Comment:
Let's remove the stack trace from the logs here, since we know it's a TimeoutException, and if it happens there's bound to be a lot of these.
```suggestion
                        LOG.warn("Timeout occurred while sending message to topic {}, it will be attempted again.", topic);
```